### PR TITLE
Support wagtail42

### PR DESCRIPTION
**Pull request by nickmoreton**
This upgrades the package for Wagtail 4.2

and includes PR: https://github.com/torchbox/wagtail-webstories/pull/19

It also drops support for Wagtail < 4.1

Todo:

- There is a failing test for wagtail 4.2 that should be OK once wagtailmedia on python 3.11 is updated

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,67 +7,164 @@ on:
   pull_request:
 
 # Current configuration:
-# - django 2.2, python 3.6, wagtail 2.5, sqlite
 # - django 2.2, python 3.7, wagtail 2.7, postgres
```
Comment (victoriachan) [inline .github/workflows/test.yml:10]: Should we get rid of < Wagtail 4.1 ?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -7,67 +7,164 @@ on:
...
+          - python: 3.7
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.1,<4.2"
+            database: "postgresql"
             experimental: false
           - python: 3.7
-            django: "Django>=2.2,<2.3"
-            wagtail: "wagtail>=2.7,<2.8"
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.2,<5.0"
```
Comment (victoriachan) [inline .github/workflows/test.yml:34]: Suggestion: Again, I was wondering if we should limit to <4.3 instead of <5.0? What do you think?

**Review by victoriachan — APPROVED** [review]
Looks good and installs well when I tried locally. Just a non blocking question/suggestion about testing up to wagtail <4.3 instead of <5.

File: .github/workflows/test.yml (hunk 2 above)
Comment (nickmoreton) [reply .github/workflows/test.yml:34]: We've been using <5.0 on others. As far as I know there will never be a 4.3 as the next release will be 5.0
I see what you are saying though.

File: setup.py [hunk 3]
```diff
@@ -37,7 +37,7 @@
         "testing": [
             "responses>=0.12,<1",
             "Pillow>=4.0.0,<9.0.0",
-            "wagtailmedia>=0.6",
+            "wagtailmedia>=0.6,<0.13",
```
Comment (nickmoreton) [inline setup.py]: wagtailmedia >= 0.13 has and issue raised that is the cause of a testing failure here: https://github.com/torchbox/wagtail-webstories/actions/runs/4313681735/jobs/7525779129

I've pinned it here so it's more obvious it needs attention for wagtail >= 5.0

**Review by victoriachan — APPROVED** [review]
OK, happy with it. Thank you!

**Comment by nickmoreton** [comment]
Superseded by https://github.com/torchbox/wagtail-webstories/pull/22

**Pull request closed without merging**